### PR TITLE
task: automatically update atlas sdk go

### PR DESCRIPTION
**Pull request by wtrocki**
## Proposed changes

I have noticed that we are behind with sdk updates for the CLI.
This PR introduces a job that will run once a week and automatically reduce boilerplate work.

I also fixed a bug in the Golang proxy that improperly handles such large major versions :)

## Verification

https://github.com/mongodb/mongodb-atlas-cli/pull/2394

### CODE REVIEW

**Review by blva — APPROVED** [review]
LGTM! wondering if we'd like to set this up in terraform and cfn?

**Comment by wtrocki** [comment]
> LGTM! wondering if we'd like to set this up in terraform and cfn?

On my radar after testing with CLI

File: .github/workflows/autoupdate-sdk.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Update SDK
+on:
+ schedule:
+   - cron: 30 8 * * MON
```
Comment (gssbzn) [inline .github/workflows/autoupdate-sdk.yaml:4]: we prefer Tuesdays as we usually have bank holidays on Monday

Comment (wtrocki) [reply .github/workflows/autoupdate-sdk.yaml:4]:
```suggestion
   - cron: 30 8 * * TUE
```

File: .github/workflows/autoupdate-sdk.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        id: verify-changed-files
+        with:
+          files: |
+             ./internal/**/*
+      - uses: peter-evans/create-pull-request@v5
+        if: steps.verify-changed-files.outputs.files_changed == 'true'
+
+        with:
+          title: "APIBot: SDK update"
+          commit-message: "fix: update go SDK"
```
Comment (gssbzn) [inline .github/workflows/autoupdate-sdk.yaml:34]: let's follow dependabot
```suggestion
          commit-message: "build(deps): bump go.mongodb.org/atlas-sdk"
```

File: .github/workflows/autoupdate-sdk.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+          delete-branch: true
+          branch: sdk-update
```
Comment (gssbzn) [inline .github/workflows/autoupdate-sdk.yaml:36]:
```suggestion
          branch: atlas-sdk-update
```

File: Makefile [hunk 4]
```diff
@@ -193,10 +193,11 @@ check-library-owners: ## Check that all the dependencies in go.mod has a owner i
...
 .PHONY: update-atlas-sdk
 update-atlas-sdk: ## Update the atlas-sdk dependency
-	@echo "==> Updating SDK to latest major version"
-	gomajor get go.mongodb.org/atlas-sdk/v20230201001@latest
+	$(eval LATEST_SDK_RELEASE := $(shell curl -sSfL -X GET  https://api.github.com/repos/mongodb/atlas-sdk-go/releases/latest | jq -r '.tag_name' | cut -d '.' -f 1 ))
+	@echo "==> Updating SDK to latest major version $(LATEST_SDK_RELEASE)"
+	gomajor get go.mongodb.org/atlas-sdk/$(LATEST_SDK_RELEASE)@latest
 	go mod tidy
-	@echo "==> Done, remember to update build/ci/library_owners.json"
+	 @echo "==> Done, remember to update build/ci/library_owners.json"
```
Comment (gssbzn) [inline Makefile:200]:
```suggestion
	@echo "==> Done, remember to update build/ci/library_owners.json"
```

File: Makefile [hunk 5]
```diff
@@ -193,10 +193,11 @@ check-library-owners: ## Check that all the dependencies in go.mod has a owner i
 
 .PHONY: update-atlas-sdk
 update-atlas-sdk: ## Update the atlas-sdk dependency
-	@echo "==> Updating SDK to latest major version"
-	gomajor get go.mongodb.org/atlas-sdk/v20230201001@latest
+	$(eval LATEST_SDK_RELEASE := $(shell curl -sSfL -X GET  https://api.github.com/repos/mongodb/atlas-sdk-go/releases/latest | jq -r '.tag_name' | cut -d '.' -f 1 ))
```
Comment (gssbzn) [inline Makefile:196]: this is probably a good time to move to bash

**Review by blva — APPROVED** [review]

File: scripts/update-sdk.sh [hunk 6]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 MongoDB Inc
```
Comment (gssbzn) [inline scripts/update-sdk.sh:3]:
```suggestion
# Copyright 2023 MongoDB Inc
```

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**